From 1fad6361aedd19c44d5f1d85be01ba07299b3fb6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kim=20Gr=C3=A4sman?= <kim.grasman@gmail.com>
Date: Sun, 7 Jul 2024 20:51:49 +0200
Subject: [PATCH] Remove brittle offload_openmp driver testcase

This testcase kind-of proved that our driver drops OpenMP/NVPTX jobs,
but it was heavily dependent on the target environment.

I'd only executed it on my own machine, and our CI environment, neither
of which have any OpenMP toolchains installed.

If such a toolchain _is_ installed the failure diagnostics will not
match, and the test will fail.

Since we don't have any control of the environment for these tests, and
no way to isolate them, remove the test.
---
 tests/driver/offload_openmp.c | 26 --------------------------
 1 file changed, 26 deletions(-)
 delete mode 100644 tests/driver/offload_openmp.c

diff --git a/tests/driver/offload_openmp.c b/tests/driver/offload_openmp.c
deleted file mode 100644
index 2625369c9..000000000
--- a/tests/driver/offload_openmp.c
+++ /dev/null
@@ -1,26 +0,0 @@
-//===--- offload_openmp.c - test input file for IWYU ----------------------===//
-//
-//                     The LLVM Compiler Infrastructure
-//
-// This file is distributed under the University of Illinois Open Source
-// License. See LICENSE.TXT for details.
-//
-//===----------------------------------------------------------------------===//
-
-// Check that IWYU ignores the extra offload compiler job produced when
-// compiling for OpenMP.
-
-// IWYU_ARGS: -fopenmp -fopenmp-targets=nvptx64 -nocudalib
-
-// This first diagnostic only happens because I don't have an nvptx64 toolchain
-// on my machine -- this test should maybe be conditional on the presence of
-// such a toolchain.
-// IWYU~: Executable ".*" doesn't exist!; consider passing it via '-march'
-
-// IWYU~: ignoring offload job for device toolchain: openmp
-
-/**** IWYU_SUMMARY(0)
-
-(tests/driver/offload_openmp.c has correct #includes/fwd-decls)
-
-***** IWYU_SUMMARY */
